maykin_common/throttling.py

```diff
@@ -2,32 +2,71 @@
 Provide mixins for throttling/rate limiting in views.
 
 Depends on ``django-axes``.
+
+.. todo:: Decouple from django-axes - make IP address getter function configurable.
 """
 
+import warnings
 from collections.abc import Container
 from time import time
 from typing import Literal
 
 from django.core.cache import caches
+from django.core.cache.backends.base import BaseCache
 from django.core.exceptions import ImproperlyConfigured, PermissionDenied
-from django.http import HttpRequest
+from django.http import HttpRequest, HttpResponse, HttpResponseBase
 
 from axes.helpers import get_client_ip_address
 
+ONE_MINUTE = 60
+ONE_HOUR = ONE_MINUTE * 60
+
 
 class ThrottleMixin:
     """
     A very simple throttling implementation with, hopefully, sane defaults.
 
-    You can specifiy the amount of visits (throttle_visits) a view can get,
-    for a specific period (in seconds) throttle_period.
+    You can specifiy the amount of visits (``throttle_visits``) a view can get,
+    for a specific period (in seconds) ``throttle_period``.
     """
 
-    # n visits per period (in seconds)
     throttle_visits = 100
-    throttle_period = 60**2  # in seconds
-    throttle_403 = True
+    """
+    Number of allowed visits in the specified period.
+    """
+
+    throttle_period = ONE_HOUR
+    """
+    Period/time window (in seconds) in which the visits are counted.
+
+    Visits older than this window are discarded.
+    """
+
     throttle_name = "default"
+    """
+    Identifier for the throttle, used in the cache key.
+    """
+
+    throttle_cache = "default"
```

Comment:
it's a bit silly to have to override a method for what is otherwise a simple setting/attribute, so I added this.